### DISCUSSION TITLE
Failing when publishing to GitHub Packages

### DISCUSSION THREAD

**Discussion by mcleonard** [community/community#30949] [Packages] [answered]
I'm trying to build a JavaScript library from a private repo in my org and publish it to GitHub Packages. Followed all the steps in the docs. Made a PAT, authenticating successfully as far as I can tell. My `package.json` looks like:
```json
{
  "name": "@myorg/my-project",
  "collaborators": [
    "Mat Leonard <leonard.mat@gmail.com>"
  ],
  "version": "0.1.0",
  "files": [
    "my_project_bg.wasm",
    "my_project.js",
    "my_project_bg.js",
    "my_project.d.ts"
  ],
  "module": "my_project.js",
  "types": "my_project.d.ts",
  "sideEffects": false,
  "repository": "https://github.com/myorg/my_project.git"
}
```
I have a `.npmrc` file next to `project.json` like
```
@myorg:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=token...
```

I repeatedly get this error:
```
notice Publishing to https://npm.pkg.github.com
44 http fetch PUT 400 https://npm.pkg.github.com/@myorg%2fmy-project 476ms
45 timing command:publish Completed in 647ms
46 verbose stack HttpErrorGeneral: 400 Bad Request - PUT https://npm.pkg.github.com/@myorg%2fmy-project
```
It just says Bad Request without more information to help me. As far as I can tell, I'm doing everything correctly. Is there a way to get more information so I can tell what's going on? Or do I need to do anything special on the org level? Maybe there's something I need to do because the repo and package are private?

**Comment by chadqueen** [comment]
I have encountered the same the past 2 hours. I have tried many variations of config after the instructions in the [npm registry dev guide](URL 'https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-to-github-package-registry') lead to the 400 Bad Request response during publishing but I always received the same error response.

**Comment by nickaguilarh** [comment]
I have also been getting the same error for almost 6 hours now. I have tried with private and public repositories. Maybe it's a GitHub thing? 
Please let me know if you solve the problem.

**Reply by mcleonard** [reply]
Yep, it's been an all day thing for me too. I think something is broken on GitHub's side if multiple people are having the same problem.

**Reply by nickaguilarh** [reply]
Still failing for me. Is it working for you now?

**Reply by carlmlane** [reply]
I'm getting this exact issue since last night through this morning (been experiencing for ~15 or so hours)

**Reply by metahimadri** [reply]
I have lost whole day yesterday trying fixing this. Something is seriously wrong. ( It is working with one of my account but not with the other account)

**Comment by tuleyko** [comment]
Same for me. Tried different configurations and no success

**Comment by carlmlane** [comment]
I'm experiencing this issue as well.

**Comment by patricksalmeida** [comment]
Same problem for me.

**Comment by shovon** [comment]
Exactly the same problem on my end. I did everything I could. No dice.

Below is a perfect copy+paste of something that I posted on Stack Overflow.

----

I have no idea what I'm doing wrong.

I have followed every possible tutorial on the Internet that involves both publishing directly from the command line by invoking `npm login --scope=@shovon --registry=https://npm.pkg.github.com`, and also publishing from a GitHub action workflow.

Always fails. I always get the same 400 error.

Out of desperation, I attempted to follow a [tutorial](https://levelup.gitconnected.com/private-npm-packages-in-github-package-registry-fbfda43acab3) to the letter, but no dice.

This is the output from the log after npm publish, and looking at the log output (assuming it even helps).

```
0 verbose cli /Users/shovon/.nvm/versions/node/v18.2.0/bin/node /Users/shovon/.nvm/versions/node/v18.2.0/bin/npm
1 info using npm@8.9.0
2 info using node@v18.2.0
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/npmrc Completed in 0ms
6 timing config:load:builtin Completed in 0ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/Users/shovon/projects/ts-npm-example/.npmrc Completed in 2ms
10 timing config:load:project Completed in 4ms
11 timing config:load:file:/Users/shovon/.npmrc Completed in 1ms
12 timing config:load:user Completed in 1ms
13 timing config:load:file:/Users/shovon/.nvm/versions/node/v18.2.0/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:credentials Completed in 0ms
17 timing config:load:setEnvs Completed in 1ms
18 timing config:load Completed in 9ms
19 timing npm:load:configload Completed in 9ms
20 timing npm:load:mkdirpcache Completed in 0ms
21 timing npm:load:mkdirplogs Completed in 0ms
22 verbose title npm publish
23 verbose argv "publish"
24 timing npm:load:setTitle Completed in 7ms
25 timing config:load:flatten Completed in 1ms
26 timing npm:load:display Completed in 1ms
27 verbose logfile logs-max:10 dir:/Users/shovon/.npm/_logs
28 verbose logfile /Users/shovon/.npm/_logs/2022-08-28T16_10_41_326Z-debug-0.log
29 timing npm:load:logFile Completed in 4ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 21ms
33 verbose publish [ '.' ]
34 silly logfile start cleaning logs, removing 1 files
35 silly logfile done cleaning log files
36 notice
37 notice 📦  @shovon/ts-npm-example@0.1.0
38 notice === Tarball Contents ===
39 notice 1.1kB LICENSE                                      
39 notice 3.4kB README.md                                    
39 notice 61B   dist/index.d.ts                              
39 notice 206B  dist/index.js                                
39 notice 242B  dist/ts-npm-example.cjs.development.js       
39 notice 440B  dist/ts-npm-example.cjs.development.js.map   
39 notice 169B  dist/ts-npm-example.cjs.production.min.js    
39 notice 328B  dist/ts-npm-example.cjs.production.min.js.map
39 notice 192B  dist/ts-npm-example.esm.js                   
39 notice 498B  dist/ts-npm-example.esm.js.map               
39 notice 1.1kB package.json                                 
39 notice 143B  src/index.ts                                 
40 notice === Tarball Details ===
41 notice name:          @shovon/ts-npm-example                  
41 notice version:       0.1.0                                   
41 notice filename:      @shovon/ts-npm-example-0.1.0.tgz        
41 notice package size:  3.7 kB                                  
41 notice unpacked size: 7.9 kB                                  
41 notice shasum:        a190bac8221697e99c37157f9fb94da822bc392c
41 notice integrity:     sha512-ZYAhozMKOz7Yx[...]yPtB03TtHLCcA==
41 notice total files:   12                                      
42 notice
43 notice Publishing to https://npm.pkg.github.com/shovon
44 http fetch PUT 400 https://npm.pkg.github.com/shovon/@shovon%2fts-npm-example 314ms
45 timing command:publish Completed in 3474ms
46 verbose stack HttpErrorGeneral: 400 Bad Request - PUT https://npm.pkg.github.com/shovon/@shovon%2fts-npm-example
46 verbose stack     at /Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/node_modules/npm-registry-fetch/lib/check-response.js:93:15
46 verbose stack     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
46 verbose stack     at async publish (/Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/node_modules/libnpmpublish/lib/publish.js:43:12)
46 verbose stack     at async otplease (/Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/lib/utils/otplease.js:3:12)
46 verbose stack     at async Publish.exec (/Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/lib/commands/publish.js:114:7)
46 verbose stack     at async module.exports (/Users/shovon/.nvm/versions/node/v18.2.0/lib/node_modules/npm/lib/cli.js:78:5)
47 verbose statusCode 400
48 verbose pkgid @shovon/ts-npm-example@0.1.0
49 verbose cwd /Users/shovon/projects/ts-npm-example
50 verbose Darwin 21.5.0
51 verbose node v18.2.0
52 verbose npm  v8.9.0
53 error code E400
54 error 400 Bad Request - PUT https://npm.pkg.github.com/shovon/@shovon%2fts-npm-example
55 verbose exit 1
56 timing npm Completed in 3509ms
57 verbose code 1
58 error A complete log of this run can be found in:
58 error     /Users/shovon/.npm/_logs/2022-08-28T16_10_41_326Z-debug-0.log
```

**Comment by shovon** [comment]
I have just created a ticket to GitHub support. Hopefully, that grabs their attention.

**Comment by walleautomacao** [comment]
Same problem for me

**Reply by sanjeevh93** [reply]
@walleautomacao 
Could you please check if this is working now?

**Comment by chadqueen** [comment]
My package just published to the registry today after being unable to Friday. I did publish a package via GITHUB_TOKEN in a workflow in between, so maybe that affected the subsequent PAT submission. I recommend trying your submissions again and seeing if it works.

If your PAT based npm publish still fails, maybe try to do a GITHUB_TOKEN workflow based publish in a public repository using this template: https://github.com/github-packages-examples/npm-publish

**Reply by nlwillia** [reply]
I'm getting a different response today than I was yesterday.  Yesterday, it was the 400 reported above, today it's a 403.  Same PAT (has `write:packages`), tried with both `.npmrc` and `npm login`.  I *can* `npm publish` a test package to my public personal repo, but I have `Admin` on the private org repo so there shouldn't be a permissions issue there.  (This is PAT based as [their docs](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-npm-registry#authenticating-with-a-personal-access-token) seem to indicate is supported.)

```
npm ERR! code E403
npm ERR! 403 403 Forbidden - PUT https://npm.pkg.github.com/@org-name%2fpackage-name - Permission permission_denied: create_package
npm ERR! 403 In most cases, you or one of your dependencies are requesting
npm ERR! 403 a package version that is forbidden by your security policy, or
npm ERR! 403 on a server you do not have access to.
```

**Reply by nickaguilarh** [reply]
I'm also using a PAT with this permissions and it's working. 
`repo, write:packages`

This is the config I'm using:
```      # Setup .npmrc file to publish to GitHub Packages
      - uses: actions/setup-node@v3
        with:
          node-version: '18.x'
          registry-url: 'https://npm.pkg.github.com'
          scope: ${{ github.repository_owner }}
      - run: npm ci
      - name: Publish to GitHub Package Registry
        run: npm publish
        env:
          NODE_AUTH_TOKEN: ${{ secrets.PAT }}```

**Reply by AlexisNoeMunoz** [reply]
I have the same error than you @nlwillia, could you fix it?

**Reply by nlwillia** [reply]
@AlexisNoeMunoz you best bet is to use a PAT.  I never had reliable results with GITHUB_TOKEN.  Also don't forget to pass the `registry-url` property.  It won't generate the `.npmrc` file without it.

**Reply by AlexisNoeMunoz** [reply]
Hi, thank you, I am using PAT with the `.npmrc` file and it just doesn't work, I have admin role in the repository and my token has write:packages permission.

```
registry=https://registry.yarnpkg.com/

@arbolfinanciero:registry=https://npm.pkg.github.com
//npm.pkg.github.com/:_authToken=TOKEN
always-auth=true
```

**Comment by nickaguilarh** [comment]
My package is working right now. Thanks for the update.

**Comment by shovon** [comment]
Things seem to be working, now.

**Comment by sanjeevh93** [comment, marked as answer]
Hi:wave: from Github Packages Team here,

The issue was limited to packages published to new namespaces and is fixed as of yesterday. 
Please check if this is working.

**Reply by mcleonard** [reply]
Works now, thank you!

**Comment by Akash1134** [comment]
GitHub Actions and Packages are [now separate categories](https://github.com/orgs/community/discussions/48284). While both of their features play well together, they are separate products and we want to give each of them their own space for questions and feedback. Keeping that in mind, this discussion has been moved over from the [Actions](https://github.com/orgs/community/discussions/categories/actions) category.

**Comment by ShrayBans** [comment]
Was hitting this error and didn't realize that my organization name (ORGANIZATION/REPO) and my scope @ORGANIZATION/PACKAGE had to be the same. Then it started working! Not convenient, but fixed my problem!

**Reply by gailmiller-lysn** [reply]
THANK YOU, dead set legend, I had missed a '-' in the package name compared to the way github creates the default npm scope!  Typo! :D

**Reply by ronsrebro** [reply]
wow. I searched for hours, creating 15 different PAT until I saw your comment. 
Yes! make sure your package scope = your org name.
